### PR TITLE
[Keymap] Update Helix:five_rows OLED code

### PR DESCRIPTION
**Pull request by mtei**
## Description

Update helix rev2 and rev3 keymap 'five_rows'.

*  Stop using snprintf() in keymaps/five_rows/oled_display.c.
    The binary size becomes 1350 bytes smaller.
* add matrix scan rate display to OLED.
* add matrix_output_unselect_delay.c.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/helix/rev2/keymaps/five_rows/matrix_output_unselect_delay.c [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/helix/rev2/keymaps/five_rows/matrix_output_unselect_delay.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Comment (mtei) [reply keyboards/helix/rev2/keymaps/five_rows/matrix_output_unselect_delay.c:17]: Done.

**Review by drashna — APPROVED** [review]

File: keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c [hunk 2]
```diff
@@ -154,12 +223,10 @@ void render_status(void) {
...
-    char led[40];
-    snprintf(led, sizeof(led), "\n%s  %s  %s",
-             (host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? "NUMLOCK" : "       ",
-             (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? "CAPS" : "    ",
-             (host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? "SCLK" : "    ");
-    oled_write(led, false);
+    oled_write_P(PSTR("\n"), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? PSTR("NUMLOCK  ") : PSTR("         "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? PSTR("CAPS  ") : PSTR("      "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? PSTR("SCLK  ") : PSTR("      "), false);
```
Comment (fauxpark) [inline keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c:229]:
```suggestion
    led_t led_state = host_keyboard_led_state();
    oled_write_P(PSTR("\n"), false);
    oled_write_P(led_state.num_lock ? PSTR("NUMLOCK  ") : PSTR("         "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAPS  ") : PSTR("      "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCLK  ") : PSTR("      "), false);
```

File: keyboards/helix/rev2/keymaps/five_rows/oled_display.c [hunk 3]
```diff
@@ -154,12 +223,10 @@ void render_status(void) {
...
-    char led[40];
-    snprintf(led, sizeof(led), "\n%s  %s  %s",
-             (host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? "NUMLOCK" : "       ",
-             (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? "CAPS" : "    ",
-             (host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? "SCLK" : "    ");
-    oled_write(led, false);
+    oled_write_P(PSTR("\n"), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? PSTR("NUMLOCK  ") : PSTR("         "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? PSTR("CAPS  ") : PSTR("      "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? PSTR("SCLK  ") : PSTR("      "), false);
```
Comment (fauxpark) [inline keyboards/helix/rev2/keymaps/five_rows/oled_display.c:229]:
```suggestion
    led_t led_state = host_keyboard_led_state();
    oled_write_P(PSTR("\n"), false);
    oled_write_P(led_state.num_lock ? PSTR("NUMLOCK  ") : PSTR("         "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAPS  ") : PSTR("      "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCLK  ") : PSTR("      "), false);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c (hunk 2 above)
Comment (mtei) [reply keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c:229]: I think it's not a bad idea to get the state of each LED, such as `led_state.num_lock`.

However, I found a potential problem in `quantum/led.h`.
The `led_t` declaration is not portable code as far as I know.
Therefore, the following code in `tmk_core/common/host.c` is also not portable.
```c
led_t host_keyboard_led_state(void) { return (led_t)host_keyboard_leds(); }
```

Try compiling the code below.
```
$ cat bit_fields_test.c
#include "led.h"

led_t status1 = { .num_lock = 1 };
led_t status2 = { .raw = (1<<USB_LED_NUM_LOCK) };
```
In the following compilation, status1 and status2 have the same value.
```
$ arm-none-eabi-gcc -mlittle-endian -S bit_fields_test.c
$ grep -A1 ^status  bit_fields_test.s
status1:
	.byte	1
--
status2:
	.byte	1
```

In the following compilations status1 and status2 will not have the same value.
```
$ arm-none-eabi-gcc -mbig-endian -S bit_fields_test.c
$ grep -A1 ^status  bit_fields_test.s                
status1:
	.byte	128
--
status2:
	.byte	1
```

Comment (fauxpark) [reply keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c:229]: AVR is little-endian, and it seems all the currently supported ARM MCUs default to little-endian, so I don't see this as a pressing issue just yet.

Comment (mtei) [reply keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c]: Done.

File: keyboards/helix/rev2/keymaps/five_rows/oled_display.c (hunk 3 above)
Comment (mtei) [reply keyboards/helix/rev2/keymaps/five_rows/oled_display.c]: Done.

**Pull request merged**